lib/editor/atto/plugins/kalturamedia/yui/build/moodle-atto_kalturamedia-button/moodle-atto_kalturamedia-button-debug.js

```diff
@@ -53,43 +54,56 @@ Y.namespace('M.atto_kalturamedia').Button = Y.Base.create('button', Y.M.editor_a
                 callback: this._kalturamedia
             });
     },
-    _kalturamedia: function(){
+    _kalturamedia: function() {
         this._currentSelection = this.get('host').getSelection();
         if (this._currentSelection === false) {
             return;
         }
+
+        var embedButton = "<button id='KalturaMediaSubmit' disabled='disabled' hidden='hidden' style='display: none'>" + M.util.get_string('embedbuttontext', COMPONENTNAME) + "</embed>";
+
+        var height = 580;
+        var width = 1100;
+        var panelHeight = height + 90;
+        var panelWidth = width + 23 + 'px';
+        width += 'px';
+
+        if (Y.UA.ipod !== 0 || Y.UA.ipad !== 0 || Y.UA.iphone !== 0 || Y.UA.android !== 0 || Y.UA.mobile !== null) {
+            panelWidth = '80%';
+            width = '100%';
+        }
+
+        var iframe = "<iframe id='panelcontentframe' height='" + height + "px' width='" + width + "' src='"+this._getIframeURL()+"'></iframe>";
+        var panelbodycontent = iframe + embedButton;
         
-        var dialogue = this.getDialogue({
-            headerContent: M.util.get_string('popuptitle', COMPONENTNAME),
-            focusAfterHide: true,
-            width: '1112px',
-            focusOnShowSelector: SELECTORS.URLINPUT
-        });
+        if (Y.UA.ipod !== 0 || Y.UA.ipad !== 0 || Y.UA.iphone !== 0) {
+            // This outer div will constrain the iframe from overlapping over its content region on iOS devices.
+            panelbodycontent = "<div id='panelcontentframecontainer'>" + iframe + embedButton + "</div>";
+        }    
+
+        if(this.panel !== null) {
+            this.panel.destroy();
+            this.panel = null;
+        }
 
-        var iframe = Y.Node.create('<iframe></iframe>');
-        // We set the height here because otherwise it is really small. That might not look
-        // very nice on mobile devices, but we considered that enough for now.
-        iframe.setStyles({
-            height: '600px',
-            border: 'none',
-            width: '100%'
+        this.panel = new Y.Panel({
+            srcNode : Y.Node.create('<div id="dialog" />'),
```

Comment:
@muli (again, not familiar with the plugin context...) unless 'dialiog' is a mandatory id name, I wouldn't use it.
Id should be a unique identifier and 'dialog' pretty common name. If there is no need for uniqueness, why not use class name instead ?